src/python/WMCore/WorkQueue/WMBSHelper.py

```diff
@@ -516,19 +526,19 @@ def _createFilesInDBSBuffer(self):
         It does the actual job of creating things in DBSBuffer
 
         """
-        if len(self.dbsFilesToCreate) == 0:
+        if not self.dbsFilesToCreate:
             # Whoops, nothing to do!
             return
 
-        dbsFileTuples = []
+        dbsFileTuples = set()
         dbsFileLoc = []
         dbsCksumBinds = []
-        locationsToAdd = []
+        locationsToAdd = set()
```

Comment:
Although this doesn't affect much performance increase, (only ~30 locations), keep this as set might be better.